### PR TITLE
Sils/res

### PR DESCRIPTION
**Pull request by sils1297**
CC @jayvdb I think you were hitting this issue as well

### CODE REVIEW

File: coalib/results/result_actions/ApplyPatchAction.py [hunk 1]
```diff
@@ -16,12 +16,15 @@ def is_applicable(result, original_file_dict, file_diff_dict):
...
+            nonempty_patches = False
+            for filename, diff in result.diffs.items():
+                if diff and (filename not in file_diff_dict or
+                             diff + file_diff_dict[filename] !=
+                             file_diff_dict[filename]):
+                    nonempty_patches = True
 
-            return True
+            return nonempty_patches
         except ConflictError:
```
Comment (adtac) [inline coalib/results/result_actions/ApplyPatchAction.py]: can this Exception still be raised now that you've removed the `__add__` call?

Comment (sils1297) [inline coalib/results/result_actions/ApplyPatchAction.py]: the addition still happens, `+` is an invocation o `__add__`, now we use the result and compare it with the original patch!

Comment (adtac) [inline coalib/results/result_actions/ApplyPatchAction.py]: ugh, I never knew `+` calls `__add__` :(

Comment (adtac) [inline coalib/results/result_actions/ApplyPatchAction.py]: But it all makes sense now

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/2834/commits/87b07cc65ff087e9746a758a3cae4fe4c129b7b8 https://github.com/coala/coala/pull/2834/commits/42526dec0257d672c4b73a1aa4c1abf859faa9a1

**Comment by adtac** [comment]
@rultor merge

**Pull request merged**